robot-core/src/main/java/org/obolibrary/robot/template/Template.java

```diff
@@ -0,0 +1,1852 @@
+      throws Exception {
+    this.ioHelper = ioHelper;
+    this.checker = checker;
+
+    tableRows = new ArrayList<>();
+    templates = new ArrayList<>();
+    headers = new ArrayList<>();
+    axioms = new HashSet<>();
+
+    // Add the contents of the tableRows
+    addTable(name, rows);
+
+    // Set all the new labels used in the template
+    // Other labels can be found from the input ontology with the checker
+    addLabels();
+    createParser();
+  }
+
+  /**
+   * Generate an OWLOntology based on the rows of the template.
+   *
+   * @return new OWLOntology
+   * @throws Exception on issue parsing rows to axioms or creating new ontology
+   */
+  public OWLOntology generateOutputOntology() throws Exception {
+    return generateOutputOntology(null);
+  }
+
+  /**
+   * Generate an OWLOntology with given IRI based on the rows of the template.
+   *
+   * @return new OWLOntology
+   * @throws Exception on issue parsing rows to axioms or creating new ontology
+   */
+  public OWLOntology generateOutputOntology(String outputIRI) throws Exception {
+    processRows();
+
+    OWLOntologyManager manager = OWLManager.createOWLOntologyManager();
+    OWLOntology outputOntology;
+    if (outputIRI != null) {
+      IRI iri = IRI.create(outputIRI);
+      outputOntology = manager.createOntology(iri);
+    } else {
+      outputOntology = manager.createOntology();
+    }
+
+    manager.addAxioms(outputOntology, axioms);
+
+    return outputOntology;
+  }
+
+  /**
+   * Given the name of a table and a list of rows, first validate the headers and template strings.
+   * Then, get the location of important columns (e.g. IDs and labels). Finally, add all template
+   * rows to the object.
+   *
+   * @param name name of table
+   * @param rows list of rows (lists)
+   * @throws Exception on malformed template
+   */
+  private void addTable(String name, List<List<String>> rows) throws Exception {
+    // Get and validate headers
+    headers = rows.get(0);
+    templates = rows.get(1);
+    if (headers.size() != templates.size()) {
+      throw new Exception(
+          String.format(columnMismatchError, name, headers.size(), templates.size()));
+    }
+
+    for (int column = 0; column < templates.size(); column++) {
+      String template = templates.get(column);
+
+      // If the template is null or the column is empty, skip this column
+      if (template == null) {
+        continue;
+      }
+      template = template.trim();
+      if (template.isEmpty()) {
+        continue;
+      }
+
+      // Validate the template string
+      if (!TemplateHelper.validateTemplateString(template)) {
+        throw new Exception(
+            String.format(unknownTemplateError, name, column + 1, headers.get(column), template));
+      }
+
+      // Get the location of important columns
+      switch (template) {
+        case "ID":
+          idColumn = column;
+          break;
+        case "LABEL":
+        case "A rdfs:label":
```

Comment:
This is cheating. What if the user chooses a different prefix or label for `rdfs:label`? It should still work.

Reply:
The `LABEL` column is used to be able to specify terms by label. If a user is using a different AP, it will still be added to the class. But how would we know if a given AP is what the user considers a "label"?

Reply:
`LABEL` is a special keyword. That’s fine. I’m saying that if we accept `A label` then I don’t think that it shouldn’t be handled as a special case: we should check each `A foo` to see if the name resolves to `rdfs:label`, based on previously declared prefixes and labels.

Reply:
Changed to only `case "LABEL"`